Expensify/App
# PR 2: Made EmojiPicker a Singleton Component

### Details

- This is the second PR for the same tagged issue, which makes the EmojiPicker a singleton component by keeping a common ref. Refer the comment https://github.com/Expensify/App/issues/3245#issuecomment-1025441734

- One change is that earlier we used to set props like `updatePreferredSkinTone`, etc to the Popover child component. Because we're setting reference for the `Popover`, I've had to move these props to `EmojiPickerMenu` separately to `index.js` and `index.native.js`
- Additionally, changed the `Popover` with `PopoverWithMeasuredContent` for `EmojiPicker`
### Fixed Issues

$ https://github.com/Expensify/App/issues/3245

### Tests

1. Tested that Emojis work fine as before
2. Tested the skintone selection for Compose
3. Tested the frequently used emojis for Compose


### QA Steps

1. Go to any chat
2. Click on EmojiPicker
3. Select the Emoji and enter
4. You can change the Skintone and ensure that it works fine as before.


### Tested On


### Screenshots


#### Web

<img width="438" alt="web-emoji-picker-with-singleton" src="https://user-images.githubusercontent.com/3069065/152618825-9bea8e66-c992-42e7-a1d4-79671f4a6539.png">

#### Mobile Web

<img width="410" alt="mweb-emoji-picker-with-singleton" src="https://user-images.githubusercontent.com/3069065/152618845-8a1b24a6-c256-4098-8a2e-81942bed5b24.png">

#### Desktop

<img width="495" alt="desktop-emoji-picker-with-singleton" src="https://user-images.githubusercontent.com/3069065/152618875-9f5c0cb5-db5d-4fdb-935f-8c09511ad021.png">

#### iOS

<img width="359" alt="ios-emoji-picker-with-singleton" src="https://user-images.githubusercontent.com/3069065/152619464-89683869-c9f0-4b08-9d48-0453fa033c6d.png">

#### Android

<img width="344" alt="android-emoji-picker-with-singleton" src="https://user-images.githubusercontent.com/3069065/152619162-789c0aa9-6d19-4781-aa0b-8f398153a20e.png">
